aml-development/ozp-backend
# feat(listing): track listing visit counts per user

AMLNG-652

**Description**     
AML 3.0 Storefront has a section for frequently visited applications.  An endpoint needs to be created to allow the storage of viewed/launch applications in the user's profile info. 

**Acceptance Criteria**   
1. End point available to track number of visits to an application by a user
2. End point available to remove a listing from frequently viewed listings

**How to test code**    
Pull `listing_view_counts` from `ozp-backend`
Pull `listing_view_counts` from `frontend`

_Testing the Backend_
Update the DB with sample data: `make dev run`

Test endpoint to get frequently visited listings and compare it with actual listing visit counts.  If you are logged into swagger (ie as bigbrother), you can use:
```
GET /api/profile/{profile_pk}/listingvisits/frequent/
GET /api/profile/{profile_pk}/listingvisits/
```

Test endpoint to increment a listing visit count (you only need to supply a listing ID in the payload).
``` 
POST /api/profile/{profile_pk}/listingvisits/increment/

Sample payload: { "listing": { "id": 1 }}
```

Test endpoint to clear a listing visit count (you only need to supply a listing ID in the payload).
``` 
POST /api/profile/{profile_pk}/listingvisits/clear/

Sample payload: { "listing": { "id": 1 }}
```

Ensure that if you can view the listing, it will increment the count by one every time the API is called, and that if the clear API is called, the count is set to 0.  Also, you can check the other API calls to make sure the frequently visited listings update based on changes to the visit count.

_Testing the Frontend_
Go to the homepage.

Ensure that frequently visited listings successfully loads.

Go to a listing page.  Ensure that the endpoint to increment the listing view gets called.

After going through a series of listing pages, ensure that the storefront frequently visited listings section updates with expected results.

**Please Review**     
@aml-development/ozp-backend-team    

**Checklist**
